@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -533,15 +554,16 @@ export class TextElementsRenderer {
      * @returns Whether the text renderer is initialized.
      */
     private initialize(textElementsAvailable: boolean): boolean {
-        if (this.m_initState === InitState.Uninitialized && textElementsAvailable) {
-            this.m_initState = InitState.Initializing;
-            this.invalidateCache(); // Force cache update after initialization.
+        if (!this.initialized && !this.initializing && textElementsAvailable) {
             this.initializeDefaultAssets();
-            this.initializeTextCanvases().then(() => {
-                this.m_initState = InitState.Initialized;
+            this.m_initPromise = this.initializeTextCanvases().then(() => {
+                this.m_initialized = true;
+                this.m_initPromise = undefined;
+                this.invalidateCache(); // Force cache update after initialization.
+                this.m_viewUpdateCallback();
             });
         }
-        return this.m_initState === InitState.Initialized;
+        return this.initialized;
```

Comment:
Maybe it could be simpler just to return Promise instead of bool, and get rid of waitInitialized() method, of course it changes the usage too. Depends from the usage scope. Where do you want to use waitLoaded() (and thus waitInitialized)? I can not find this...

Reply:
Initialization is handled completely within the class and done lazily only when there's some text to render. That's why initialize() is a private method.
MapView just blindly calls placeText() and renderText() on each frame, that way we keep client code from having to know too much about the state of the text renderer. Initialize is called from placeText() when needed, and there we don't want to wait for initialization.

The wait methods are there mainly to simplify testing. Synchronizing the test with the class initialization makes the test more understandable and less error prone.

Reply:
> Initialization is handled completely within the class and done lazily only when there's some text to render. That's why initialize() is a private method.
> MapView just blindly calls placeText() and renderText() on each frame, that way we keep client code from having to know too much about the state of the text renderer. Initialize is called from placeText() when needed, and there we don't want to wait for initialization.

That part was clear for me, sorry maybe I have asked to generic question.

>
> The wait methods are there mainly to simplify testing. Synchronizing the test with the class initialization makes the test more understandable and less error prone.

This is something I am missing - do we have `waitLoaded()` method used somewhere now (tests)? Could not find it. Maybe it would be nice to write some test code using it.

Reply:
I have a commit on top of this one adding unit testing for TextElementsRenderer, which uses this wait methods:

https://github.com/heremaps/harp.gl/commit/a47fb656967006963e8a6966720d02cd77343eb1

Reply:
Ok got ya.